spectrocloud/resource_cluster_profile.go

```diff
@@ -64,17 +64,22 @@ func resourceClusterProfile() *schema.Resource {
 				Optional: true,
 			},
 			"cloud": {
-				Type:     schema.TypeString,
-				Default:  "all",
-				Optional: true,
-				ForceNew: true,
+				Type:         schema.TypeString,
+				Default:      "all",
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"all", "aws", "azure", "gcp", "vsphere", "openstack", "maas", "nested", "baremetal", "eks", "aks", "edge", "edge-native", "libvirt", "tencent", "tke", "coxedge", "generic", "gke"}, false),
+				ForceNew:     true,
+				Description: "A cloud provider type supported in Palette, Supported cloud types are - `all, aws, azure, gcp, vsphere, openstack, maas, nested, baremetal, eks, aks, edge, edge-native, libvirt, tencent, tke, coxedge, generic and gke`," +
+					"If a cloud value is `all` then type must be `add-on`. Otherwise it should be a corresponding cloud type.",
 			},
 			"type": {
 				Type:         schema.TypeString,
 				Optional:     true,
 				Default:      "add-on",
 				ValidateFunc: validation.StringInSlice([]string{"add-on", "cluster", "infra", "system"}, false),
-				ForceNew:     true,
+				Description: "Dictates the layers that can be configured in the cluster profile. Valid types - `cluster` - Corresponding value in UI is Full, " +
+					"`infra` - Corresponding value in UI is Infrastructure, `add-on` and `system` - For system profile refer UI documentation.",
```

Comment:
```suggestion
				Description: "Specify the cluster profile type to use. Allowed values are `cluster`, infra`, `add-on`, and `system`. These values map to the following User Interface (UI) labels. Use the value ' cluster ' for a **Full** cluster profile." + "For an Infrastructure cluster profile, use the value `infra`; for an Add-on cluster profile, use the value `add-on`." + "System cluster profiles can be specified using the value `system`. To learn more about cluster profiles, refer to the [Cluster Profile](https://docs.spectrocloud.com/cluster-profiles) documentation.",
```
A suggestion for how to improve the message. I also added a link to the docs.